src/vsphere_cpi/lib/cloud/vsphere/file_provider.rb

```diff
@@ -43,7 +43,7 @@ def upload_file_to_url(url, body, headers)
     private
 
     def fetch_file_from_datastore_via_datacenter(datacenter_name, datastore_name, path)
-      url ="https://#{@vcenter_host}/folder/#{path}?dcPath=#{URI.escape(datacenter_name)}" +
+      url ="https://#{@vcenter_host}/folder/#{URI.escape(path)}?dcPath=#{URI.escape(datacenter_name)}" +
```

Comment:
We have to be careful here. I'm not sure if `path` when passed into this function is already URL escaped. If it is then we'll end up double escaping this. In at least one case:

```ruby
    def get_current_env(vm, datacenter_name)
      logger.info("Getting current agent env from vm '#{vm.name}' in datacenter '#{datacenter_name}'")
      cdrom = @client.get_cdrom_device(vm)
      env_iso_folder = env_iso_folder(cdrom)
      return unless env_iso_folder

      datastore = cdrom.backing.datastore
      datastore_pattern = Regexp.escape(datastore.name)
      result = env_iso_folder.match(/\[#{datastore_pattern}\] (.*)/)
      raise Bosh::Clouds::CloudError.new("Could not find matching datastore name '#{datastore.name}'") unless result
      env_path = result[1]

      contents = @file_provider.fetch_file_from_datastore(datacenter_name, datastore, "#{env_path}/env.json")
      raise Bosh::Clouds::CloudError.new("Unable to load env.json from '#{env_path}/env.json'") unless contents

      JSON.load(contents)
    end
```

This makes me nervous that we may break existing behavior.

The same goes for the other changes of a similar nature in this file.

Reply:
@ktchen14 I have tested and the `path` here is not escaped yet. Since we are not supporting non-ASCII letters now, I will roll back it ( to make minimum change to CPI). And when we trying to  support non-ASCII letters, we change it.